air/src/execution_step/air/compare_matchable/comparator.rs

```diff
@@ -16,57 +16,50 @@
 
 use crate::execution_step::air::ExecutionResult;
 use crate::execution_step::execution_context::ExecutionCtx;
-use crate::execution_step::utils::resolve_ast_variable;
+use crate::execution_step::utils::resolve_ast_variable_wl;
 use crate::JValue;
 
 use air_parser::ast;
-use air_parser::ast::MatchableValue;
+use air_parser::ast::AIRValue;
 
 pub(crate) fn are_matchable_eq<'ctx>(
-    left: &MatchableValue<'_>,
-    right: &MatchableValue<'_>,
+    left: &AIRValue<'_>,
+    right: &AIRValue<'_>,
     exec_ctx: &'ctx ExecutionCtx<'_>,
 ) -> ExecutionResult<bool> {
-    use MatchableValue::*;
+    use AIRValue::*;
 
     match (left, right) {
         (InitPeerId, InitPeerId) => Ok(true),
-        (InitPeerId, matchable) => compare_matchable(
-            matchable,
-            exec_ctx,
-            make_string_comparator(exec_ctx.init_peer_id.as_str()),
-        ),
-        (matchable, InitPeerId) => compare_matchable(
+        (InitPeerId, matchable) | (matchable, InitPeerId) => compare_matchable(
             matchable,
             exec_ctx,
             make_string_comparator(exec_ctx.init_peer_id.as_str()),
         ),
 
-        (Literal(left_name), Literal(right_name)) => Ok(left_name == right_name),
-        (Literal(value), matchable) => compare_matchable(matchable, exec_ctx, make_string_comparator(value)),
-        (matchable, Literal(value)) => compare_matchable(matchable, exec_ctx, make_string_comparator(value)),
-
-        (Boolean(value), matchable) => compare_matchable(matchable, exec_ctx, make_bool_comparator(value)),
-        (matchable, Boolean(value)) => compare_matchable(matchable, exec_ctx, make_bool_comparator(value)),
-
-        (Number(value), matchable) => compare_matchable(matchable, exec_ctx, make_number_comparator(value)),
-        (matchable, Number(value)) => compare_matchable(matchable, exec_ctx, make_number_comparator(value)),
+        (LastError(left_error), LastError(right_error)) => Ok(left_error == right_error),
+        (LastError(error), matchable) | (matchable, LastError(error)) => {
+            compare_matchable(matchable, exec_ctx, make_string_comparator(&error.to_string()))
```

Comment:
And here are you comparing `LastErrorPath::to_string` with an actual value? 
```
pub enum LastErrorPath {
    // %last_error%.instruction
    Instruction,
    // %last_error%.msg
    Message,
    // %last_error%.peer_id
    PeerId,
    // %last_error%
    None,
}
```
I'm confused :)

Reply:
Fixed